Performing checks and cleaning our final annotated data

In [1]:
import pandas as pd
import numpy as np

data_path = 'joe_finished_data.csv'
joe = pd.read_csv(data_path)

sinan = pd.read_csv('sinan_finished_data.csv')

joe

Unnamed: 0  annotation_id  annotator                           artist  \
0             0            102          2                       Rich Amiri   
1             1            103          2                      Lil Calorie   
2             2            104          2                          Safaree   
3             3            105          2                   Vanilla Taurus   
4             4            106          2                      ProduceKyro   
..          ...            ...        ...                              ...   
995        1395           1197          2  Kris Kristofferson & Brenda Lee   
996        1396           1198          2             The Osborne Brothers   
997        1397           1199          2                    Reba McEntire   
998        1398           1200          2                     Aaron Watson   
999        1399           1201          2                The Rankin Family   

                      created_at    id  lead_time  \
0    2024-07-23T17:04:57.737680Z   101    249.679   
1    2024-07-23T17:07:46.230072Z   102    158.014   
2    2024-07-23T17:16:38.949418Z   103    530.711   
3    2024-07-23T17:18:32.407446Z   104    107.249   
4    2024-07-23T17:23:07.854164Z   105    267.982   
..                           ...   ...        ...   
995  2024-08-16T14:43:42.426689Z  1496     45.114   
996  2024-08-16T14:43:51.727822Z  1497      8.881   
997  2024-08-16T14:44:04.081827Z  1498     11.919   
998  2024-08-16T14:44:22.625580Z  1499     18.040   
999  2024-08-16T14:44:37.864252Z  1500     14.825   

                                                lyrics      tag  \
0    [Intro]\r\nKeep it up, Sharkboy\r\n\r\n[Chorus...      rap   
1    I’m just an extra\r\nNobody text yeah\r\nLeave...      rap   
2    (Papi Chuloh)\r\n\r\n[?]\r\n\r\nStuntgang\r\n\...      rap   
3    [Chorus] x2\r\n\r\nI-I-I-\r\nUp in California\...      rap   
4    [Intro]\r\n(You, you, you, you)\r\n(You, you, ...      rap   
..                                                 ...      ...   
995  Her long legs looked like trouble\r\nAs she sa...  country   
996  Ruby, Ruby honey, are you mad at your man?\r\n...  country   
997  [Verse 1]\r\nLove's the only thing I know in t...  country   
998  [Verse 1]\r\nYellow dog bus, blacktop highway\...  country   
999  [Verse 1]\r\nThese days there's too much infor...  country   

                                       title                   updated_at  \
0                                      Relax  2024-07-23T17:04:57.737680Z   
1                                      Extra  2024-07-23T17:07:46.230072Z   
2                               The Most Fun  2024-07-23T17:16:38.949418Z   
3                                 California  2024-07-23T17:18:32.407446Z   
4                              Free Thinking  2024-07-23T17:23:07.854164Z   
..                                       ...                          ...   
995  The Bigger the Fool the Harder the Fall  2024-08-16T14:43:42.426689Z   
996                         Ruby Are You Mad  2024-08-16T14:43:51.727822Z   
997                               A New Love  2024-08-16T14:44:04.081827Z   
998                           Touchdown Town  2024-08-16T14:44:22.625580Z   
999                    Nothing Like An Ocean  2024-08-16T14:44:37.864252Z   

     x_coordinate  y_coordinate  year  
0            -0.4           0.6  2022  
1            -0.7          -0.4  2020  
2             0.5           0.5  2015  
3            -0.3          -0.3  2018  
4             0.7          -0.2  2021  
..            ...           ...   ...  
995          -0.5           0.4  1982  
996          -0.3           0.6  1959  
997          -0.6          -0.3  1989  
998           0.5           0.8  2020  
999           0.5          -0.3  2007  

[1000 rows x 14 columns]

Dropping unused columns

In [2]:
joe.drop(columns=['Unnamed: 0', 'annotation_id', 'annotator', 'created_at', 'id', 'lead_time', 'updated_at'], inplace=True)

Even spread of genres as seen here

In [3]:
joe['tag'].value_counts()

tag
rap        200
rb         200
rock       200
pop        200
country    200
Name: count, dtype: int64

In [4]:
joe.columns

Index(['artist', 'lyrics', 'tag', 'title', 'x_coordinate', 'y_coordinate',
       'year'],
      dtype='object')

In [5]:
sinan

index                                              title      tag  \
0         0                                              Relax      rap   
1         1                                              Extra      rap   
2         2                                       The Most Fun      rap   
3         3                                         California      rap   
4         4                                      Free Thinking      rap   
...     ...                                                ...      ...   
1495   1495                                     Whiskey Chasin  country   
1496   1496                            Smoke In a Bar Acoustic  country   
1497   1497                                   Away In A Manger  country   
1498   1498                                  Sourwood Mountain  country   
1499   1499  If Youre Ever In Oklahoma - Live At INTRUST Ba...  country   

              artist  year  x_coordinate  y_coordinate  
0         Rich Amiri  2022         -0.40          0.70  
1        Lil Calorie  2020         -0.80         -0.30  
2            Safaree  2015          0.70          0.85  
3     Vanilla Taurus  2018         -0.70          0.55  
4        ProduceKyro  2021          0.25          0.65  
...              ...   ...           ...           ...  
1495    Joe Stampley  2015           NaN           NaN  
1496    Travis Tritt  2022           NaN           NaN  
1497    Brad Paisley  2006           NaN           NaN  
1498     Pete Seeger  1965           NaN           NaN  
1499     Eric Church  2017           NaN           NaN  

[1500 rows x 7 columns]

In [6]:
sinan.isnull().sum()

index             0
title             0
tag               0
artist            0
year              0
x_coordinate    497
y_coordinate    497
dtype: int64

Removing unannotated values (null) from sinans data

In [7]:
sinan.dropna(axis=0, inplace=True)

In [8]:
sinan

index                  title      tag                artist  year  \
0         0                  Relax      rap            Rich Amiri  2022   
1         1                  Extra      rap           Lil Calorie  2020   
2         2           The Most Fun      rap               Safaree  2015   
3         3             California      rap        Vanilla Taurus  2018   
4         4          Free Thinking      rap           ProduceKyro  2021   
...     ...                    ...      ...                   ...   ...   
1396   1396       Ruby Are You Mad  country  The Osborne Brothers  1959   
1397   1397             A New Love  country         Reba McEntire  1989   
1398   1398         Touchdown Town  country          Aaron Watson  2020   
1399   1399  Nothing Like An Ocean  country     The Rankin Family  2007   
1400   1400            Ill Be Gone  country          Sarah Jarosz  2020   

      x_coordinate  y_coordinate  
0            -0.40          0.70  
1            -0.80         -0.30  
2             0.70          0.85  
3            -0.70          0.55  
4             0.25          0.65  
...            ...           ...  
1396         -0.70          0.30  
1397         -0.70         -0.30  
1398          0.70          0.85  
1399         -0.30         -0.70  
1400         -0.70          0.30  

[1003 rows x 7 columns]

In [9]:
sinan.drop(columns='index', inplace=True)

In [10]:
sinan

title      tag                artist  year  \
0                     Relax      rap            Rich Amiri  2022   
1                     Extra      rap           Lil Calorie  2020   
2              The Most Fun      rap               Safaree  2015   
3                California      rap        Vanilla Taurus  2018   
4             Free Thinking      rap           ProduceKyro  2021   
...                     ...      ...                   ...   ...   
1396       Ruby Are You Mad  country  The Osborne Brothers  1959   
1397             A New Love  country         Reba McEntire  1989   
1398         Touchdown Town  country          Aaron Watson  2020   
1399  Nothing Like An Ocean  country     The Rankin Family  2007   
1400            Ill Be Gone  country          Sarah Jarosz  2020   

      x_coordinate  y_coordinate  
0            -0.40          0.70  
1            -0.80         -0.30  
2             0.70          0.85  
3            -0.70          0.55  
4             0.25          0.65  
...            ...           ...  
1396         -0.70          0.30  
1397         -0.70         -0.30  
1398          0.70          0.85  
1399         -0.30         -0.70  
1400         -0.70          0.30  

[1003 rows x 6 columns]

In [11]:
sinan['tag'].value_counts()

tag
rb         201
rock       201
pop        201
rap        200
country    200
Name: count, dtype: int64

In [12]:
sinan_to_compare = sinan[['year', 'title', 'artist', 'tag']]

In [13]:
joe_to_compare = joe[['year', 'title', 'artist', 'tag']]

In [14]:
# concatenate dataframes - joins together
combined_df = pd.concat([joe_to_compare, sinan_to_compare])

# Drop duplicates and keep only the unique rows
unique_rows = combined_df.drop_duplicates()

# The number of common rows
num_common_rows = len(combined_df) - len(unique_rows)

print(f"Number of common rows: {num_common_rows}")

# This cant be right, further investigation needed

Number of common rows: 975


975 common rows is definitely not correct, looking into this further

In [15]:
# ~ reverses a boolean. So this shows rows that are not in the other df
non_matching_rows = sinan_to_compare[~sinan_to_compare['title'].isin(joe_to_compare['title'])]

print(len(non_matching_rows))  

non_matching_rows

# shows that the possible issue with row matching is titles not artists.

28


year                                              title  \
29    2021                                 ‚Äã‚Äãthelifestyle   
194   2020                                              C√òLD   
200   2021                                       Lemonade Boy   
305   1984                                    It’s the Love   
307   2022                                  Fallin’ in Love   
324   2020                                ‚Äãput in the grave   
371   1991                      Forever My Lady K-Ci’s Drum   
411   2007  Cherry Live At Shepherd’s Bush Empire London...   
433   1971                I’m Afraid The Masquerade is Over   
463   2021                                   Go Don’t Leave   
490   2021  C H E B . الشاب - طربوش دالورد...   
500   2020                                             Me Too   
624   2010                               We’re Not the Same   
635   1984                    What’s Love Got to Do with It   
675   2014               Smashing Pumpkins’s “Daydream”   
800   2018                                     Six Feet Under   
909   2021                          ‚Äãi need to say goodbye.   
960   2020               ‚Äãlove nwantiti German Frizzo Remix   
1073  2021                                            ‚Äãilwy   
1077  2020                                        ‚Äãoblivion   
1094  2019  Mafumafu „Åæ„ŵ„Åæ„ŵ  - I want to be a Girl Â...   
1098  2018                     애걔 FOR English Translation   
1100  2001                                        Lovers Leap   
1230  1984                             She’s Out of My Life   
1291  2021                       That’s how much I love you   
1309  1958                     We’ll Always Have Each Other   
1371  1963                                  Marie’s Wedding   
1400  2020                                        Ill Be Gone   

                                 artist      tag  
29                            Eli juggz      rap  
194                           777Arcane      rap  
200                       JordyBoy (QC)      rap  
305                       Carl Anderson       rb  
307                            RUSSELL!       rb  
324                               czyby       rb  
371                              Jodeci       rb  
411                       Amy Winehouse       rb  
433                        David Porter       rb  
463               BluPrint (Girl Group)       rb  
490         Genius English Translations       rb  
500                            MC Penis       rb  
624                      Enjoy The View     rock  
635                         Tina Turner     rock  
675                         Mr. Mendola     rock  
800                          Cloverdale     rock  
909                                Mive      pop  
960                                CKay      pop  
1073                           frocious      pop  
1077                              Coals      pop  
1094        Genius English Translations      pop  
1098        Genius English Translations      pop  
1100                     Creeper Lagoon      pop  
1230                      Willie Nelson  country  
1291                          Jean wolf  country  
1309                       Kenny Rogers  country  
1371  The Clancy Brothers & Tommy Makem  country  
1400                       Sarah Jarosz  country

This shows the lower than expected common rows is due to randomly added characters

In [16]:
# this shows the artists, and therefore entries, do exist in my df, they just have symbol typos in title.
# example
for artist in joe_to_compare['artist']:
    if artist == 'Eli juggz':
        print(artist)

Eli juggz


All artists with randomly added chars, mostly do appear, just need to search by artist

In [17]:
non_matching_rows = sinan_to_compare[~sinan_to_compare['artist'].isin(joe_to_compare['artist'])]

print(len(non_matching_rows))  

non_matching_rows

# these entries are the outliers from sinans df!!! Tested on the code chunk above, artist did not print.


5


year           title          artist      tag
200   2021    Lemonade Boy   JordyBoy (QC)      rap
500   2020          Me Too        MC Penis       rb
800   2018  Six Feet Under      Cloverdale     rock
1100  2001     Lovers Leap  Creeper Lagoon      pop
1400  2020     Ill Be Gone    Sarah Jarosz  country

Shows the indexes of the songs actually not in joes data

In [18]:
# Find rows in joe_to_compare that do not have a matching title in sinans df
non_matching_rows = joe_to_compare[~joe_to_compare['title'].isin(sinan_to_compare['title'])]

print(len(non_matching_rows))

non_matching_rows


25


year                                              title  \
29   2021                                     ​​thelifestyle   
159  2014                                    Judges Criteria   
194  2020                                               CØLD   
205  1984                                      It’s the Love   
207  2022                                    Fallin’ in Love   
224  2020                                  ​put in the grave   
271  1991                        Forever My Lady K-Ci’s Drum   
311  2007  Cherry Live At Shepherd’s Bush Empire London /...   
333  1971                  I’m Afraid The Masquerade is Over   
363  2021                                     Go Don’t Leave   
390  2021  C H E B . الشاب - طربوش دالورد - F R U I T Eng...   
424  2010                                 We’re Not the Same   
435  1984                      What’s Love Got to Do with It   
475  2014                     Smashing Pumpkins’s “Daydream”   
609  2021                            ​i need to say goodbye.   
660  2020                 ​love nwantiti German Frizzo Remix   
773  2021                                              ​ilwy   
777  2020                                          ​oblivion   
794  2019  Mafumafu まふまふ  - I want to be a Girl 女の子になりたい ...   
798  2018                         애걔 FOR English Translation   
830  1984                               She’s Out of My Life   
891  2021                         That’s how much I love you   
909  1958                       We’ll Always Have Each Other   
971  1963                                    Marie’s Wedding   
974  2018  ERP Implementation in Automobile Industry | Pr...   

                                artist      tag  
29                           Eli juggz      rap  
159                         Arobie1992      rap  
194                          777Arcane      rap  
205                      Carl Anderson       rb  
207                           RUSSELL!       rb  
224                              czyby       rb  
271                             Jodeci       rb  
311                      Amy Winehouse       rb  
333                       David Porter       rb  
363              BluPrint (Girl Group)       rb  
390        Genius English Translations       rb  
424                     Enjoy The View     rock  
435                        Tina Turner     rock  
475                        Mr. Mendola     rock  
609                               Mive      pop  
660                               CKay      pop  
773                           frocious      pop  
777                              Coals      pop  
794        Genius English Translations      pop  
798        Genius English Translations      pop  
830                      Willie Nelson  country  
891                          Jean wolf  country  
909                       Kenny Rogers  country  
971  The Clancy Brothers & Tommy Makem  country  
974                           Pridesys  country

In [19]:
# Find rows in joe_to_compare that do not have a matching title in sinans df
non_matching_rows = joe_to_compare[~joe_to_compare['artist'].isin(sinan_to_compare['artist'])]

print(len(non_matching_rows))

non_matching_rows

2


year                                              title      artist  \
159  2014                                    Judges Criteria  Arobie1992   
974  2018  ERP Implementation in Automobile Industry | Pr...    Pridesys   

         tag  
159      rap  
974  country

By searching by artist, found all outlier songs, from where annotations overlap or differ

In [20]:
sinan.drop(index=[200,500,800,1100,1400], inplace=True)

In [21]:
joe.drop(index=[159,974], inplace=True)

Reading in sams data

In [22]:
samuel = pd.read_csv('samdb.csv')

samuel

annotation_id  annotator          artist                   created_at  \
0                 2          1      Rich Amiri  2024-07-17T19:00:34.065082Z   
1                 3          1     Lil Calorie  2024-07-17T19:01:59.697045Z   
2                 4          1         Safaree  2024-07-17T19:04:19.189196Z   
3                 5          1  Vanilla Taurus  2024-07-17T19:07:07.881957Z   
4                 6          1     ProduceKyro  2024-07-17T19:13:49.366715Z   
...             ...        ...             ...                          ...   
1095            175          1    Joe Stampley  2024-07-29T18:26:21.180507Z   
1096            174          1    Travis Tritt  2024-07-29T18:25:57.100411Z   
1097            173          1    Brad Paisley  2024-07-29T18:24:37.297481Z   
1098            172          1     Pete Seeger  2024-07-29T18:23:50.329744Z   
1099            171          1     Eric Church  2024-07-29T18:23:17.321926Z   

        id  index  lead_time  \
0     1498      0    160.866   
1     1499      1     76.316   
2     1500      2    137.862   
3     1501      3     91.000   
4     1502      4    177.028   
...    ...    ...        ...   
1095  2993   1495     22.161   
1096  2994   1496     77.548   
1097  2995   1497     44.943   
1098  2996   1498     30.574   
1099  2997   1499     42.035   

                                                 lyrics      tag  \
0     [Intro]\r\nKeep it up, Sharkboy\r\n\r\n[Chorus...      rap   
1     I’m just an extra\r\nNobody text yeah\r\nLeave...      rap   
2     (Papi Chuloh)\r\n\r\n[?]\r\n\r\nStuntgang\r\n\...      rap   
3     [Chorus] x2\r\n\r\nI-I-I-\r\nUp in California\...      rap   
4     [Intro]\r\n(You, you, you, you)\r\n(You, you, ...      rap   
...                                                 ...      ...   
1095  Looks like this bad rainy weather\r\nKept ever...  country   
1096  [Verse 1]\r\nDaddies were Daddies and Mamas we...  country   
1097  [Chorus]\r\nAway in a manger, no crib for a be...  country   
1098  Chickens a-crowin' on Sourwood Mountain\r\nHey...  country   
1099  If you're ever in Oklahoma\r\nYou better move ...  country   

                                                  title  \
0                                                 Relax   
1                                                 Extra   
2                                          The Most Fun   
3                                            California   
4                                         Free Thinking   
...                                                 ...   
1095                                     Whiskey Chasin   
1096                            Smoke In a Bar Acoustic   
1097                                   Away In A Manger   
1098                                  Sourwood Mountain   
1099  If Youre Ever In Oklahoma - Live At INTRUST Ba...   

                       updated_at x_coordinate y_coordinate  year  
0     2024-07-17T19:00:34.065082Z         -0.4          0.8  2022  
1     2024-07-17T19:01:59.697045Z         -0.8         -0.2  2020  
2     2024-07-17T19:04:19.189196Z          0.5          0.4  2015  
3     2024-07-17T19:07:07.881957Z         -0.5         -0.3  2018  
4     2024-07-17T19:13:49.367717Z          0.1          0.4  2021  
...                           ...          ...          ...   ...  
1095  2024-07-29T18:26:21.180507Z         -0.9         -0.5  2015  
1096  2024-07-29T18:25:57.100411Z          0.6         -0.4  2022  
1097  2024-07-29T18:24:37.297481Z          0.2         -0.9  2006  
1098  2024-07-29T18:23:50.329744Z          0.6         -0.4  1965  
1099  2024-07-29T18:23:17.321926Z          0.2         -0.9  2017  

[1100 rows x 14 columns]

In [23]:
samuel.drop(columns=['index', 'annotation_id', 'annotator', 'created_at', 'id', 'lead_time', 'updated_at'], inplace=True)

In [24]:
samuel

artist                                             lyrics  \
0         Rich Amiri  [Intro]\r\nKeep it up, Sharkboy\r\n\r\n[Chorus...   
1        Lil Calorie  I’m just an extra\r\nNobody text yeah\r\nLeave...   
2            Safaree  (Papi Chuloh)\r\n\r\n[?]\r\n\r\nStuntgang\r\n\...   
3     Vanilla Taurus  [Chorus] x2\r\n\r\nI-I-I-\r\nUp in California\...   
4        ProduceKyro  [Intro]\r\n(You, you, you, you)\r\n(You, you, ...   
...              ...                                                ...   
1095    Joe Stampley  Looks like this bad rainy weather\r\nKept ever...   
1096    Travis Tritt  [Verse 1]\r\nDaddies were Daddies and Mamas we...   
1097    Brad Paisley  [Chorus]\r\nAway in a manger, no crib for a be...   
1098     Pete Seeger  Chickens a-crowin' on Sourwood Mountain\r\nHey...   
1099     Eric Church  If you're ever in Oklahoma\r\nYou better move ...   

          tag                                              title x_coordinate  \
0         rap                                              Relax         -0.4   
1         rap                                              Extra         -0.8   
2         rap                                       The Most Fun          0.5   
3         rap                                         California         -0.5   
4         rap                                      Free Thinking          0.1   
...       ...                                                ...          ...   
1095  country                                     Whiskey Chasin         -0.9   
1096  country                            Smoke In a Bar Acoustic          0.6   
1097  country                                   Away In A Manger          0.2   
1098  country                                  Sourwood Mountain          0.6   
1099  country  If Youre Ever In Oklahoma - Live At INTRUST Ba...          0.2   

     y_coordinate  year  
0             0.8  2022  
1            -0.2  2020  
2             0.4  2015  
3            -0.3  2018  
4             0.4  2021  
...           ...   ...  
1095         -0.5  2015  
1096         -0.4  2022  
1097         -0.9  2006  
1098         -0.4  1965  
1099         -0.9  2017  

[1100 rows x 7 columns]

In [25]:
samuel['tag'].value_counts()

tag
country    300
rap        200
rb         200
rock       200
pop        200
Name: count, dtype: int64

Need to find the 100 spare country songs

In [26]:
sam_to_compare = samuel[['year', 'tag', 'artist','title']]

In [27]:
non_matching_rows = sam_to_compare[~sam_to_compare['title'].isin(joe_to_compare['title'])]

print(len(non_matching_rows))

non_matching_rows

99


year      tag          artist  \
1000  2020  country    Sarah Jarosz   
1001  2019  country        Upchurch   
1002  2010  country  Zac Brown Band   
1003  2017  country        Brieanna   
1004  1980  country   John Anderson   
...    ...      ...             ...   
1095  2015  country    Joe Stampley   
1096  2022  country    Travis Tritt   
1097  2006  country    Brad Paisley   
1098  1965  country     Pete Seeger   
1099  2017  country     Eric Church   

                                                  title  
1000                                        Ill Be Gone  
1001                                             Fallen  
1002                                         Smoke Rise  
1003                                       Country Song  
1004              She Just Started Liking Cheatin Songs  
...                                                 ...  
1095                                     Whiskey Chasin  
1096                            Smoke In a Bar Acoustic  
1097                                   Away In A Manger  
1098                                  Sourwood Mountain  
1099  If Youre Ever In Oklahoma - Live At INTRUST Ba...  

[99 rows x 4 columns]

recognise the top entry from these non matching rows as an entry not in joes dataset so these must be correct

In [28]:
samuel = samuel.drop(index=range(1000,1100))

In [29]:
samuel['tag'].value_counts()

tag
rap        200
rb         200
rock       200
pop        200
country    200
Name: count, dtype: int64

In [30]:
# fresh comparison matrix after dropping rows

sam_to_compare = samuel[['year', 'tag', 'artist','title']]

In [31]:
non_matching_rows = sam_to_compare[~sam_to_compare['artist'].isin(sinan_to_compare['artist'])]

print(len(non_matching_rows))

non_matching_rows

2


year      tag      artist  \
159  2014      rap  Arobie1992   
974  2018  country    Pridesys   

                                                 title  
159                                    Judges Criteria  
974  ERP Implementation in Automobile Industry | Pr...

Same 2 different songs as joes data

In [32]:
samuel.drop(index=[159, 974], inplace=True)

Some additional checks due to initial errors when running data through agreement file

In [33]:
# additional checks due to errors calculating alpha

sinan_x = sinan['x_coordinate'].tolist()
sinan_y = sinan['y_coordinate'].tolist()

sam_x = samuel['x_coordinate'].tolist()
sam_y = samuel['y_coordinate'].tolist()

for i, coord in enumerate(sam_x):
    if coord == '--0.8':
        print(f"Error caused by index {i}")

sam_x[186] = '-0.8'

for i, coord in enumerate(sam_y):
    if coord == '-.0.5':
        print(f"Error caused by index {i}")


sam_y[46] = '-0.5'

samuel['x_coordinate'] = sam_x

samuel['y_coordinate'] = sam_y
 

# rounding up values to make an even number of coords, granularity was meant to be 0.1

for i, value in enumerate(sinan_x):
    if value == 0.25:
        sinan_x[i] = 0.3
    if value == 0.85:
        sinan_x[i] = 0.9
    if value == -0.55:
        sinan_x[i] = -0.6
    if value == 0.75:
        sinan_x[i] = 0.8
    if value == -0.75:
        sinan_x[i] = -0.8
    if value == 0.65:
        sinan_x[i] = 0.7
    if value == 0.19:
        sinan_x[i] = 0.2

for i, value in enumerate(sinan_y):
    if value == 0.65:
        sinan_y[i] = 0.7
    if value == 0.85:
        sinan_y[i] = 0.9
    if value == 0.55:
        sinan_y[i] = 0.6
    if value == 0.45:
        sinan_y[i] = 0.5
    if value == -0.35:
        sinan_y[i] = -0.4
    if value == -0.15:
        sinan_y[i] = -0.2


sinan['x_coordinate'] = sinan_x

sinan['y_coordinate'] = sinan_y




Error caused by index 186
Error caused by index 46


Converting to float as they are strings, for ease of agreement calculation

In [34]:
sam_x = samuel['x_coordinate'].tolist()
sam_y = samuel['y_coordinate'].tolist()

for i, value in enumerate(sam_x):
    sam_x[i] = float(value)

for i, value in enumerate(sam_y):
    sam_y[i] = float(value)

samuel['x_coordinate'] = sam_x

samuel['y_coordinate'] = sam_y


In [35]:
samuel.to_csv('final_sam.csv')

joe.to_csv('final_joe.csv')

sinan.to_csv('final_sinan.csv')

Splitting by genre to do further analysis.

In [36]:
sam_rap = samuel[samuel['tag'] == 'rap']

sinan_rap = sinan[sinan['tag'] == 'rap']

joe_rap = joe[joe['tag'] == 'rap']

sam_rb = samuel[samuel['tag'] == 'rb']

sinan_rb = sinan[sinan['tag'] == 'rb']

joe_rb = joe[joe['tag'] == 'rb']

sam_rock = samuel[samuel['tag'] == 'rock']

sinan_rock = sinan[sinan['tag'] == 'rock']

joe_rock = joe[joe['tag'] == 'rock']

sam_pop = samuel[samuel['tag'] == 'pop']

sinan_pop = sinan[sinan['tag'] == 'pop']

joe_pop = joe[joe['tag'] == 'pop']

sam_country = samuel[samuel['tag'] == 'country']

sinan_country = sinan[sinan['tag'] == 'country']

joe_country = joe[joe['tag'] == 'country']

print(len(sam_rap), len(sinan_rap), len(joe_rap))

print(len(sam_rb), len(sinan_rb), len(joe_rb))

print(len(sam_rock), len(sinan_rock), len(joe_rock))

print(len(sam_pop), len(sinan_pop), len(joe_pop))

print(len(sam_country), len(sinan_country), len(joe_country))



199 199 199
200 200 200
200 200 200
200 200 200
199 199 199


In [37]:
sam_rap.to_csv('sam_rap.csv')
sam_rb.to_csv('sam_rb.csv')
sam_rock.to_csv('sam_rock.csv')
sam_pop.to_csv('sam_pop.csv')
sam_country.to_csv('sam_country.csv')

sinan_rap.to_csv('sinan_rap.csv')
sinan_pop.to_csv('sinan_pop.csv')
sinan_country.to_csv('sinan_country.csv')
sinan_rb.to_csv('sinan_rb.csv')
sinan_rock.to_csv('sinan_rock.csv')

joe_rap.to_csv('joe_rap.csv')
joe_country.to_csv('joe_country.csv')
joe_rb.to_csv('joe_rb.csv')
joe_rock.to_csv('joe_rock.csv')
joe_pop.to_csv('joe_pop.csv')

Making plots from recorded alpha/beta statistics

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

x_beta = [0.65, 0.70, 0.67, 0.68, 0.71, 0.68]

y_beta = [0.63, 0.63, 0.57, 0.60, 0.61, 0.62]

beta = pd.DataFrame({'x_beta': x_beta, 'y_beta': y_beta}, 
                  index=['rap', 'rnb', 'pop', 'rock', 'country', 'all_genres'])

x_alpha = [0.33, 0.40, 0.35, 0.36, 0.42, 0.38]

y_alpha = [0.26, 0.27, 0.15, 0.22, 0.22, 0.25]

alpha = pd.DataFrame({'x_alpha': x_alpha, 'y_alpha': y_alpha},
                     index=['rap', 'rnb', 'pop', 'rock', 'country', 'all_genres'])


In [ ]:
# Initialize the figure
plt.figure(figsize= (8, 8))

# Plot each dataset with different markers and colors
plt.scatter(alpha['x_alpha'], alpha['y_alpha'], label='Alpha', marker='o', color='blue')
plt.scatter(beta['x_beta'], beta['y_beta'], label='Beta', marker='s', color='green')


# Adding labels for each point
for label, x, y in zip(alpha.index, alpha['x_alpha'], alpha['y_alpha']):
    plt.text(x, y, label, fontsize=9)

for label, x, y in zip(beta.index, beta['x_beta'], beta['y_beta']):
    plt.text(x, y, label, fontsize=9)

# Set title and labels
plt.title('Comparative Scatter Plot of Different Metrics')
plt.xlabel('X')
plt.ylabel('Y')

# Display the legend
plt.legend()

# Show the plot
plt.show()


In [ ]:
import matplotlib.pyplot as plt

# Data (x_alpha and y_alpha values for each genre and annotator group)
data = {
    'All Annotators': {
        'Rap': (0.32501826220852803, 0.2585011236875123),
        'Rb': (0.4038427288822729, 0.27434175763277147),
        'Pop': (0.35371895328868197, 0.15227391830381554),
        'Rock': (0.3588177911360485, 0.22732857949604013),
        'Country': (0.42006549943317906, 0.22016557357671285),
        'All Genres': (0.38017359704748643, 0.2540492918629561),
    },
    'Annotators A & B': {
        'Rap': (0.3829486384311025, 0.2508683804112024),
        'Rb': (0.4890093437073195, 0.2762733602052412),
        'Pop': (0.44545431517085177, 0.21472328426392928),
        'Rock': (0.4468663415037635, 0.30660017399818673),
        'Country': (0.5406521928763011, 0.3067811876871406),
        'All Genres': (0.4719772789637676, 0.310410985506113),
    },
    'Annotators A & C': {
        'Rap': (0.2118821294715295, 0.28763133346076486),
        'Rb': (0.29997143906319723, 0.29611788939946926),
        'Pop': (0.2566407083422384, 0.14536157247825088),
        'Rock': (0.35657228576786737, 0.23669733679682958),
        'Country': (0.3249408736847841, 0.21446501875315083),
        'All Genres': (0.2963892527149746, 0.2551509436988634),
    },
    'Annotators B & C': {
        'Rap': (0.36670301227132973, 0.22716976037797465),
        'Rb': (0.39826814291359847, 0.24981576311864007),
        'Pop': (0.3337162985729317, 0.07929765786247633),
        'Rock': (0.24522903628463022, 0.13108547873064713),
        'Country': (0.3630670617037264, 0.1174901449361172),
        'All Genres': (0.34851418681393054, 0.1866174554143054),
    }
}

# Colors for each group
colors = {
    'All Annotators': 'blue',
    'Annotators A & B': 'green',
    'Annotators A & C': 'red',
    'Annotators B & C': 'purple'
}

# Plotting
fig, axs = plt.subplots(2, 2, figsize=(10, 10))
titles = ['All Annotators', 'Annotators A & B', 'Annotators A & C', 'Annotators B & C']

# Set the same x and y axis limits for all subplots
x_limits = (0.2, 0.55)
y_limits = (0.05, 0.35)

for ax, title in zip(axs.flat, titles):
    for genre, (x_alpha, y_alpha) in data[title].items():
        ax.scatter(x_alpha, y_alpha, label=genre, color=colors[title])
        ax.text(x_alpha, y_alpha, genre, fontsize=10, ha='left', color=colors[title])
    
    ax.set_title(title)
    ax.set_xlabel('x_alpha')
    ax.set_ylabel('y_alpha')
    ax.set_xlim(x_limits)  # Set x-axis limits
    ax.set_ylim(y_limits)  # Set y-axis limits

plt.tight_layout()
plt.show()
